shared/actions/platform-specific/input-monitor.desktop.js

```diff
@@ -1,62 +1,73 @@
 // @flow
 
+const timeToConsiderActiveForAwhile = 300000
+const timeToConsiderInactive = 60000
+
 type NotifyActiveFunction = (isActive: boolean) => void
 // 5 minutes after being active, consider us inactive after
 // an additional minute of no input
 // for the purpose of marking chat messages read
 class InputMonitor {
-  active: boolean
+  active = true
   notifyActive: NotifyActiveFunction
-  activeTimeoutID: ?number
-  inactiveTimeoutID: ?number
-  constructor(notifyActiveFunction: NotifyActiveFunction) {
-    this.notifyActive = notifyActiveFunction
-    this.active = true
+  activeTimeoutID: ?TimeoutID
+  inactiveTimeoutID: ?TimeoutID
+
+  constructor() {
+    // go into listening mode again
+    window.addEventListener('focus', this.goListening)
+    window.addEventListener('blur', this.goInactive)
+    this.goListening()
   }
 
-  startActiveTimer = () => {
-    // wait 5 minutes before adding listeners
-    if (this.activeTimeoutID) window.clearTimeout(this.activeTimeoutID)
-    this.activeTimeoutID = window.setTimeout(this.goListening, 300000)
+  _clearTimers = () => {
+    if (this.activeTimeoutID) {
```

Comment:
:sheep: I don't think these `if`s are necessary, `clearTimeout` is a noop with `null`

Reply:
i think flow wanted this, lemme see